### ISSUE TITLE
Card front showing wrong moves.

### ISSUE THREAD

**Issue by haruki241** [TowelSniffer/Anki-Chess-2.0#50]
Playing this line: 1.e4 c5 2.Nf3 d6 3.Bb5+ Nd7 4.a4 Ngf6 5.Nc3 a6 6.Be2 b6 7.d4 cxd4 8.Qxd4 Bb7 9.O-O e6 10.Rd1 Qc7 11.Be3 Be7 12.Nd2 O-O 13.Nc4 d5 14.exd5 Nxd5 15.Nxd5 Bxd5 16.Bf4

As white, flip=false, handicap=0, and the picture on the front of the card is showing incorrect moves. I go to play 4. a4 and it plays the move sound, then takes back my move and wants me to play Nc3, which would be right on move 5 but then the card continues down the line without showing the pawn is on a4.

**Comment by TowelSniffer** [comment]
This is an odd bug. I found with flip true it fails at Ngf6. This is an issue with mliebelt's pgn-parser which this template uses. Though the bug only occurs as the PGN doesn't contain a space between the number and move. Typical format would have the space, so the following works:
 
1. e4 c5 2. Nf3 d6 3. Bb5+ Nd7 4. a4 Nf6 5. Nc3 a6 6. Be2 b6 7. d4 cxd4 8. Qxd4
Bb7 9. O-O e6 10. Rd1 Qc7 11. Be3 Be7 12. Nd2 O-O 13. Nc4 d5 14. exd5 Nxd5 15.
Nxd5 Bxd5 16. Bf4 *

why it half works without the spaces I have no idea, but I didn't write the parsing logic

**Comment by haruki241** [comment]
Thanks for the quick fix. I didn't realize every single line I have is missing the space. I did a quick replace of every instance of "." with ". " and I was still having an issue on a couple lines (I decided to keep the space as it seems like the standard). 

Turns out on move 4 Ngf6 is incorrect notation, as the knight on d7 is pinned by the bishop on b5, disambiguating Nf6 as Ncf6 would be illegal. The correct notation, as you mentioned is 4... Nf6. I got this line from Wesley So's 1. e4 chessable course, so it's interesting this wasn't caught during its publication.

Another interesting note is that the lichess analysis board parses this correctly and corrects the move to 4... Nf6. Anyways I corrected it and it's all fixed now, cheers. :)

**Comment by TowelSniffer** [comment]
That's interesting, thank you for the clarification. You are correct, it if the nf6 notation, just tested myself with the spaces and it's the same issue. Should have known the spaces weren't to blame considering it was working until a certain move. definitely jumped to a conclusion there.

yeah I pasted the line into nibbler, and it also had no issues with it. I might  open an issue on mliebelt's github.

**Comment by zarevnaya** [comment]
I think it's worth a mention: the lack of a + or a # on a move that is a check or checkmate respectively also breaks the parser.

**Comment by TowelSniffer** [comment]
Yeah there are a few things. One of the issues I have found is if you have a branching line for black, it needs to include blacks move number. So

1. e4 e5 (d5) 

This breaks but

1. e4 e5 (2... d5) 

This is good. 

Some of these may be fixed in later versions of the parser. Still, it definitely works better than the pgn parser I tried to write before finding mliebelt's. That thing was a mess lol. 

I will add a comment about this to the readme. If people run into issues with their pgn, try running it through a parser with standard output like lichess/chess.com
